### DISCUSSION TITLE
No input element found. Please make sure that the "inputRef" template ref is properly assigned with ion-input

### DISCUSSION THREAD

**Discussion by Isidora25** [dm4t2/vue-currency-input#388] [Q&A] [answered]
Hello 
I'm receiving this error when I try to pass inputRef into ion-input 
```
<template>
  <ion-input :ref="inputRef" label="label" name="name" :value="formattedValue"></ion-input>
</template>

<script setup>
 const { formattedValue, numberValue, inputRef, setValue, setOptions } = useCurrencyInput({
    currency: 'EUR',
    locale: 'fr',
  })
</script>
```

How can I use inputRef with ion-input?

**Comment by dm4t2** [comment]
Hi, assigning the template ref without a colon should work: `<ion-input ref="inputRef"/>`

**Reply by Isidora25** [reply]
hello @dm4t2 ! 
thank you for your response! 
But unfortunately it doesn't work without colon as well

**Reply by dm4t2** [reply]
Can you provide a minimal reproduction ([CodeSandbox](https://codesandbox.io/) or [StackBlitz](https://stackblitz.com/))? Thanks!

**Reply by Isidora25** [reply]
[Here it is](https://codesandbox.io/s/ionic-vue-3-currency-input-c7sw81?file=/src/components/IonCurrencyInput.vue)
ion currency input - doesn't work
regular input - works fine  
the error is in console if there is only ion-input

**Reply by dm4t2** [reply, marked as answer]
Thanks! The problem is that the native HTML input element of `<ion-input>` can only be retrieved with https://ionicframework.com/docs/api/input#getinputelement. This can be solved by extending the API of `useCurrencyInput` to allow a custom lookup strategy. I will open an issue for that.

**Reply by dm4t2** [reply]
see #390

**Reply by Isidora25** [reply]
@dm4t2 thank you! :bowing_woman:

**Reply by palfaro91** [reply]
Hi @Isidora25 did you end up finding a workaround? 

@dm4t2 is there a work around?

**Reply by dm4t2** [reply]
A workaround might be [intl-number-input](https://github.com/dm4t2/intl-number-input), which can be used directly with HTML input elements.
